js/2024.js

```diff
@@ -0,0 +1,155 @@
+document.addEventListener("DOMContentLoaded", function () {
+    const animationContainer = document.getElementById('animation-container');
+    const canvas = document.getElementById('animation-canvas');
+    const title = document.getElementById('title');
+    const crescendoImg = document.getElementById('Crescendoimg');
+    const bigMantis = document.getElementById('big-Mantis');
+
+    const totalFrames = 72;
+    const startTitleFrame = 33;
+    const endTitleFrame = 37;
+    const startCrescendoFrame = 33;
+    const endCrescendoFrame = 40;
+    const crescendoDissolveFrame = 66;
+    const endDissolveFrame = 72;
+
+    let lastScrollY = window.scrollY;
+    let imagesLoaded = 0;
+
+    // Create an array to preload images
+    const preloadedImages = [];
+
+    for (let i = 1; i <= totalFrames; i++) {
+        const paddedFrame = String(i).padStart(4, '0');
+        const img = new Image();
+        img.src = `/images/Note/${paddedFrame}.png`;
+        preloadedImages.push(img);
+
+        // Increment counter when each image is loaded
+        img.onload = function () {
+            imagesLoaded++;
+            if (imagesLoaded === totalFrames) {
+                // Show everything once all images are loaded
+                document.body.classList.remove('loading');
+                drawFrame(1); // Draw the initial frame
+            }
+        };
+    }
+
+    // Hide everything until all images are loaded
+    document.body.classList.add('loading');
+
+    // Function to load and draw the image on the canvas
+    function drawFrame(frame) {
+        const paddedFrame = String(frame).padStart(4, '0');
+        const img = preloadedImages[frame - 1];
+
+        const context = canvas.getContext('2d');
+        context.clearRect(0, 0, canvas.width, canvas.height);
+        context.drawImage(img, 0, 0, canvas.width, canvas.height);
+    }
+
+    // Animate title between frames 33 and 37
+    function animateTitle(frame) {
+        if (frame < startTitleFrame) {
```

Comment:
What if they're equal? Also might be cleaner to use some ternary operators to make the code a little cleaner